src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt

```diff
@@ -29,7 +31,13 @@ class AccountSettingChangeListener(project: Project) : ChangeListener(project) {
               CodyToolWindowContent.executeOnInstanceIfNotDisposed(project) {
                 removeAllChatSessions()
               }
-              CodyAgentService.getInstance(project).restartAgent(project)
+              CodyAgentService.getInstance(project).restartAgent(project).thenApply {
+                CodyToolWindowContent.executeOnInstanceIfNotDisposed(project) {
+                  UIUtil.invokeLaterIfNeeded {
```

Comment:
I'd rather invert this wrapping ➡️ 
`executeOnInstanceIfNotDisposed` inside `invokeLaterIfNeeded`

project could dispose before `invokeLaterIfNeeded` executes